sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVScan.scala

```diff
@@ -70,8 +70,8 @@ case class CSVScan(
     ExprUtils.verifyColumnNameOfCorruptRecord(dataSchema, parsedOptions.columnNameOfCorruptRecord)
     // Don't push any filter which refers to the "virtual" column which cannot present in the input.
     // Such filters will be applied later on the upper layer.
-    val actualFilters =
-      pushedFilters.filterNot(_.references.contains(parsedOptions.columnNameOfCorruptRecord))
+    val actualFilters = pushedFilters.map(_.toV1)
+      .filterNot(_.references.contains(parsedOptions.columnNameOfCorruptRecord))
```

Comment:
Hmm, why we cannot work on v2 here and other places directly? I feel it is verbose and redundant to see `toV1`, `toV2` there.

Reply:
Currently `OrcFilters`, `ParquetFilters`, `JacksonParser`, `UnivocityParser` only take v1 filters. There are actually quite some work to refactor these to make them also work with v2 filters. I prefer to have separate PRs later on for these changes.

Reply:
+1

Reply:
I see. Maybe we can create some JIRAs for these planed changes and put the JIRA number into some comments.

Reply:
Added comments. Thanks